### PR TITLE
perf(core): remove support for the deprecated `WrappedValue`

### PR DESCRIPTION
**Pull request by JoostK**
This commit removes the `WrappedValue` public API from `@angular/core`
which was deprecated in Angular 10 without replacement.

BREAKING CHANGE:

The `WrappedValue` class can no longer be imported from `@angular/core`,
which may result in compile errors or failures at runtime if outdated
libraries are used that are still using `WrappedValue`. The usage of
`WrappedValue` should be removed as no replacement is available.

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:397877953:BASE:397877981:1632183237154:845499b3).

**Review by atscott — APPROVED** [review]
Ah, good catch! Totally forgot that we deprecated that.

LGTM. As I recall, there were a couple places in g3 that used `WrappedValue` that we'll need to update.

**Comment by AndrewKushnir** [comment]
@JoostK quick update: there is a couple places in Google's codebase where we need to do some cleanup (someone relies on the `WrappedValue` symbol) before we can land the change. The cleanup is mostly mechanical, so we should be able to do that quickly.

Could you please rebase this PR and resolve merge conflicts, so we can rerun tests if needed?

Thank you.

**Comment by JoostK** [comment]
@AndrewKushnir rebased!

File: goldens/public-api/core/core.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ export class ApplicationInitStatus {
     readonly done = false;
     // (undocumented)
     readonly donePromise: Promise<any>;
-    }
```
Comment (IgorMinar) [inline goldens/public-api/core/core.md:56]: this change is odd... do you know why it was made as part of this PR? it seems like a legit fix, I'm just surprised to see it in this PR - have we updated the api extractor version?

**Review by IgorMinar — APPROVED** [review]
LGTM! This is a great cleanup! Thank you, @JoostK. ❤️ 

Reviewed-for: public-api, global-approvers

File: goldens/public-api/core/core.md (hunk 1 above)
Comment (JoostK) [reply goldens/public-api/core/core.md:56]: Yeah I was surprised by this as well. I don't know what's caused this output to be different; it is generated exactly like this by api-extractor and not touched by a formatter afterwards. I am a bit surprised how a change to the whitespace does not cause golden file failures in the first place.

**Comment by AndrewKushnir** [comment]
[Global Presubmit #2](https://test.corp.google.com/ui#id=OCL:399603694:BASE:399830055:1632968112002:4f4e9d6) (with cleanup).

**Comment by AndrewKushnir** [comment]
FYI, global presubmit with the cleanup looks good. Once all the changes in Google's codebase land, we should be able to proceed with merging this PR. Thank you.

**Comment by AndrewKushnir** [comment]
FYI I've started [one more presubmit](http://test/OCL:399603694:BASE:399956390:1633020613292:21f7bb74) to make sure there are no new regressions (should have results in ~1hr).

**Comment by AndrewKushnir** [comment]
Note to Caretaker: cleanup in g3 is completed, TGP is green for this PR. If there are any regressions reported in g3, let's see if we can fix them forward (please reach out to me). Thank you.

**Pull request closed without merging**